subsys/bluetooth/host/scan.c

```diff
@@ -524,6 +538,36 @@ void bt_hci_le_adv_ext_report(struct net_buf *buf)
 
 		evt = net_buf_pull_mem(buf, sizeof(*evt));
 
+		if (!ext_scan_buf) {
+			/* The previous report was truncated in the host.
+			 * Discard all remaining reports for that advertisement.
+			 */
+
+			if (!(evt->evt_type & BIT(5))) {
+				/* This event is the last event to be discarded.
+				 */
+				ext_scan_buf = net_buf_alloc(&ext_scan_buf_pool,
+							     K_NO_WAIT);
+				__ASSERT_NO_MSG(ext_scan_buf);
+			}
+
+			continue;
+		}
+
+		if (evt->length + ext_scan_buf->len > net_buf_max_len(ext_scan_buf)) {
+			size_to_copy = net_buf_max_len(ext_scan_buf) - evt->length;
+			truncate = true;
+		} else {
+			size_to_copy = evt->length;
+			truncate = false;
+		}
+
+		net_buf_add_mem(ext_scan_buf, buf->data, size_to_copy);
+		if (!truncate && (evt->evt_type & BIT(5))) {
```

Comment:
You have this comparison with `BIT(5)` more than one place: If I understand correctly validating against `BIT(5)` is the same as check whether the event_type data_status value is 0b01 "incomplete, more data to come", right? 

I think it makes it a lot more readable if we an enum or macros for the 3(4) different values of Data Status, extract the 2 bits from the `evt->evt_type` value and then do the comparison (similar to how `bt_iso_recv` does it).

Reply:
- [ ] Extract out the data status to a variable